public/index.html

```diff
@@ -1,129 +1,172 @@
+      return createEditorWorker();
+    },
+  };
+
+  // Parse the search string to get url parameters.
+  var search = window.location.search;
+  var parameters = {};
+  search
+    .substr(1)
+    .split('&')
+    .forEach(function(entry) {
+      var eq = entry.indexOf('=');
+      if (eq >= 0) {
+        parameters[decodeURIComponent(entry.slice(0, eq))] =
+          decodeURIComponent(entry.slice(eq + 1));
       }
+    });
 
-      const fetcher = GraphiQL.createFetcher({
-        url:
-          parameters.fetchURL ||
-          "/graphql",
-      });
-
-      const explorerPlugin = GraphiQLPluginExplorer.explorerPlugin();
-      // Render <GraphiQL /> into the body.
-      ReactDOM.render(
-        React.createElement(GraphiQL, {
-          fetcher: fetcher,
-          query: parameters.query,
-          variables: parameters.variables,
-          operationName: parameters.operationName,
-          onEditQuery: onEditQuery,
-          onEditVariables: onEditVariables,
-          onEditOperationName: onEditOperationName,
-          plugins: [explorerPlugin]
-        }),
-        document.body
+  // if variables was provided, try to format it.
+  if (parameters.variables) {
+    try {
+      parameters.variables = JSON.stringify(
+        JSON.parse(parameters.variables),
+        null,
+        2,
       );
-    </script>
-  </body>
+    } catch (e) {
+      // Do nothing, we want to display the invalid JSON as a string, rather
+      // than present an error.
+    }
+  }
+
+  // When the query and variables string is edited, update the URL bar so
+  // that it can be easily shared
+  function onEditQuery(newQuery) {
+    parameters.query = newQuery;
+    updateURL();
+  }
+
+  function onEditVariables(newVariables) {
+    parameters.variables = newVariables;
+    updateURL();
+  }
+
+  function onEditOperationName(newOperationName) {
+    parameters.operationName = newOperationName;
+    updateURL();
+  }
+
+  function updateURL() {
+    var newSearch =
+      '?' +
+      Object.keys(parameters)
+        .filter(function(key) {
+          return Boolean(parameters[key]);
+        })
+        .map(function(key) {
+          return (
+            encodeURIComponent(key) +
+            '=' +
+            encodeURIComponent(parameters[key])
+          );
+        })
+        .join('&');
+    history.replaceState(null, null, newSearch);
+  }
+
+  const fetcher = createGraphiQLFetcher({
+    url:
+      parameters.fetchURL ||
+      '/graphql',
+  });
+  const plugins = [HISTORY_PLUGIN, explorerPlugin()];
+
+  function App() {
+    return React.createElement(GraphiQL, {
+      fetcher: fetcher,
+      initialQuery: parameters.query,
+      initialVariables: parameters.variables,
+      operationName: parameters.operationName,
+      onEditQuery: onEditQuery,
+      onEditVariables: onEditVariables,
+      onEditOperationName: onEditOperationName,
+      plugins,
+    });
+  }
+
+  const container = document.getElementById('graphiql');
```

Comment:
I guess this works despite the call being before the element exists because it's `<script type="module">` and there's some imports that cause it to run a tick later?